{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "362eeae0",
   "metadata": {},
   "source": [
    "### 深度神经网络\n",
    "1. **2017, Torng and Altman, 3D deep convolution neural networks for amino acid environment similarity analysis**\n",
    "    1. 第一个用3DCNN框架来做蛋白质分析。实用深度神经网络的好处是它能够自动的提取对任务有用的特征，不需要我们对特定的热舞作出假设和前验知识。他们用3DCNN模型来区分20个氨基酸的微环境，从而达到能够预测哪一个氨基酸对某一个位置是最合适的目的。另外他们的预测统计分析提供了每一个氨基酸被另一个氨基酸替代的的概率星系。根据这些信息得到的替换矩阵，也通过实验证明他们包含更多的信息。另外他们也提供了一种可视化技术，能够将每一个原子对最终决策的贡献可视化。他们的成功说明了用深度神经网络来做蛋白质结构分析是可行的。\n",
    "2. **2018, deep generative models of genetic variation capture the effects of mutations**\n",
    "    1. 用DeepSequence, a probabilistic model to predict the effects of mutations across a variety of deep mutational scanning experiments\n",
    "    2. it is possible to capture higher-order, context-dependent constraints in biological sequences via latent variable models with nonlinear dependencies.\n",
    "    3. <font color=red>用Bayesian deep latent variable model预测突变效果，能够揭露隐藏结构和相互关系</font>\n",
    "3. **2020, Shroff et al., Discovery of Novel Gain-of-Function mutations Guided by Structure-Based Deep Learning**\n",
    "    1. 也用3DCNN框架来识别出功能获得性的突变来增加蛋白质功能。他们在Torng and Altman的模型的基础上，他们做了一些改变来增加深度神经网络和氨基酸周围化学成分的联系。他们加入了氢原子，局部电荷，溶解可及性的相关信息。另外他们修改了数据集以此免除掉了一些误差。并且，增加了半胱氨酸和甲硫氨酸的频率来提高准确率。\n",
    "4. **2020, Li et al., Predicting changes in protein thermodynamic stability upon point mutation with deep 3D convlutional neural networks**\n",
    "    1. we introduce ThermoNet, a deep, 3D-convolutional neural network(3D-CNN) designed for structure-based prediction of $\\Delta \\Delta G$ prediction methods.\n",
    "    2. the results suggest that 3DCNNs can model the complex, non-linear interactions perturbed by mutations, directly from biophysical properties of atoms\n",
    "5. **2021, Pancotti et al., A Deep-Learning Sequence-Based Method to Predict Protein Stability Changes Upon Genetic Variations**\n",
    "6. **2022, Hongyuan Lu et al., Machine learning-aided engineering of hydrolases for PET depolymerization**\n",
    "    1. 也用了3DCNN网络来预测突变。他们通过此方法得到的FAST-PETase能在（30-50）度的反应条件下，针对51种低结晶度的PET产品包装，预处理过的高结晶度PET塑料瓶，以及化纤织物等，均能直接产生快速高效的解聚效果。\n",
    "    \n",
    "# 用处\n",
    "### mutation site prediction\n",
    "1. **2021, Ikebe et al., Enzyme modification using mutation site prediction method for enhancing the regioselectivity of substrate reaction sites**\n",
    "    1. takes conformational data for docking poses of an enzyme and a substrate as input and automatically generate a ranked list of mutation sites to destabilize docking poses for byproducts while maintaining those for target products in silico.\n",
    "    2. can distinguish differences of substrate structures and the reaction sites, and can accurately predict mutation sites to enhance regioselectivity without selection by directed evolution screening.\n",
    "\n",
    "2. **Schreiber et al., 2020, Ledidi: Designing genomic edits that induce functional activity**\n",
    "    * Ledidi, an approach that treats the design of genomic edits as an optimization problem where the goal is to produce the desired output from a predictive model. \n",
    "    * use Gumbel-Softmax reparameterization trick to solve the discrete nature\n",
    "    * the objective function contains two parts\n",
    "        1. the distance between the original sequence and the proposed edited sequence (the sequence loss)\n",
    "        2. the distance between the model output using the edited sequence and the desired output (the functional loss)\n",
    "        3. the second term is weighted by $\\lambda$ to either encourage a small number of edits when small or a closer match to the desired output when large\n",
    "    * The main idea is about how to reverse the predictive model to find a minimal set of mutations to achieve desirable targetsm\n",
    "### protein function prediction\n",
    "* 蛋白质-蛋白质相互作用 (protein-protein interactions, PPIs)\n",
    "* 药物-靶点相互作用 (Drug-Target Interactions, DTIs)\n",
    "* stability\n",
    "* thermostability \n",
    "5. **2022, Deep learning program to predict protein functions based on sequence information**\n",
    "    1. It applied sequence segmentatin during the sequence feature extraction process, by a convolution neural network, to train the regional sequence patterns and their relationship\n",
    "    2. FUTUSA successfully predicted the impact of point mutations on phenylalanine hydroxylase, which is responsible for an inherited metabolic desease PKU.\n",
    "    3. this deep-learning program can be used as the first-step tool for characterizing newly identified or poorly studied protein\n",
    "    4. 模型包含四个部分\n",
    "        1. embedding layers: \n",
    "        2. feature extraction: \n",
    "        3. dense layers\n",
    "        4. scoring steps\n",
    "    5. 用了average precision, F1-score, area under receiver operating characteristic curve, area under precision-recall curve and Matthews correlation coefficient 来评估模型\n",
    "    5. 8-12\n",
    "6. **2020, Cai et al., an integrated deep learning model for protein function prediction**\n",
    "    * SDN2GO applies convolutional neural networks to learn and extract features from sequences, protein domains and known PPI networks, and then utilizes a weight classifier to integrate these features and achieve accurate predictions of Gene Ontology terms.\n",
    "    * learned from the NLP to process domain information and pre-trained a deep learning sub-model to extract the comprehensive features of domains.\n",
    "7. **2017, Kulmanov et al., predicting protein functions from sequence and interactions using a deep ontology-aware classifier**\n",
    "8. **2019, Hoehndorf, improved protein function prediction from sequence**\n",
    "9. **2020, Zhou et al., predicting gene ontology annotations of proteins via graph convolutional network**\n",
    "10. **陈成,基于深度学习的蛋白质功能预测研究**\n",
    "    * 提出基于集成残差卷积神经网络(Ensemble Residual Convolution Neural Network)的蛋白质-蛋白质相互作用预测方法\n",
    "    * 提出基于深度神经网络(deep neural network, DNN)的药物-靶点相互作用预测方法。\n",
    "\n",
    "# 输入分类\n",
    "### structured based\n",
    "1. **2011, Wainreb et al., A single recorded mutation aids in predicting the effects of other mutations in the same amino acid site**\n",
    "2. **2006, Parthiban et al., Prediction of protein stability upon point mutations**\n",
    "3. **2018, Pucci et al., Quantification of biases in predictions of protein stability changes upon mutations.**\n",
    "4. **2020, Li et al., Predicting changes in protein thermodynamic stability upon point mutation with deep 3D convlutional neural networks**\n",
    "5. **2011, Kellogg et al., Role of conformational sampling in computing mutation-induced changes in protein structure and stability.**\n",
    "### Sequence based\n",
    "1. **2021, Pancotti et al., A Deep-Learning Sequence-Based Method to Predict Protein Stability Changes Upon Genetic Variations**\n",
    "    1. the first convolutional neural network to predict protein stability changes relying solely on the protein sequence.\n",
    "    2. able to incorporate into its architecture the antisymmetry property.\n",
    "2. **2015, Fariselli et al., Predicting the impact of non-synonymous variations on protein stability from sequence**\n",
    "3. **2019, Montanucci et al., An untrained method for the prediction of protein stability changes upon single and multiple point variations**\n",
    "4. **2021, Li et al., SAAFEC-SEQ: A Sequence-Based Method for Predicting the Effect of Single Point Mutations on Protein Thermodynamic Stability**\n",
    "###  both\n",
    "1. **2005, Capriotti et al., Predicting stability changes upon mutation from the protein sequence or structure.**\n",
    "2. **2006, Cheng et al., Prediction of protein stability changes for single-site mutations using support vector machines.**\n",
    "    1. use SVM to predict protein stability changes for single amino acid mutations leveraging both sequence and structural information.\n",
    "    2. evaluate using cross-validation methods\n",
    "    3. the model can accurately predict protein stability changes using primary sequence information only.\n",
    "    4. 作者不用神经网络用SVM的原因是 (1)不受local minima的影响 （2）已经有完善的svm算法\n",
    "        * 但是现在深度神经网络的发展，这些问题可以被解决，效果能比SVM更好\n",
    "    5. 数据来源\n",
    "        1. 通过remove site-specific redundancy 得到redundancy-reduced dataset\n",
    "        2. use dataset S1615 compiled by Capriotti et al.\n",
    "        3. S1615 is extracted from the ProTherm database\n",
    "        4. the dataset includes 1615 single site mutations obtained from 42 different proteins. Each mutation has six attributes: PDB code, mutation, solvent accessibility, pH value, temperature, and energy change.\n",
    "        5. remove redundant examples to get SR1496.\n",
    "        6. 在S1615的基础上除掉其他的redundant，得到了SR1135\n",
    "        7. 在S1615的基础上用UniqueProt去除同类蛋白质和去除掉多余的突变，得到SR1023\n",
    "        \n",
    "\n",
    "# 值得细看的论文\n",
    "1. **2020, Cai et al., an integrated deep learning model for protein function prediction**\n",
    "\n",
    "# 一些重要概念\n",
    "1. **Enzyme modification using mutation site prediction method for enhancing the regioselectivity of substrate reaction sites**\n",
    "    * activity, stability, and substrate selectivity are major considerations when modifying enzymes for enzymatic reactions for industrial use. \n",
    "2. **Prediction of Protein Stability Changes for Single-Site Mutations Using Support Vector Machines**\n",
    "    * 单个氨基酸突变的稳定性改变有四种方法\n",
    "        * physical potential approach\n",
    "        * statistical potential approach\n",
    "        * empirical potential approach\n",
    "        * machine learning approach\n",
    "    \n",
    "\n",
    "# 其他论文\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "76875793",
   "metadata": {},
   "source": [
    "### 有用的论文\n",
    "1. **2020, Jing et al., Amino acid encoding methods for protein sequence: a comprehensive review and assessment**"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "337913ac",
   "metadata": {},
   "source": [
    "随着深度学习在其他领域的成功，3DCNN也被用来从蛋白质3D结构中提取信息来预测位点突变，并且取得了成功。[11,12,13]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aeadfceb",
   "metadata": {},
   "source": [
    "# 总结\n",
    "随着深度神经网络在其他方面的成功，很多人也把深度模型应用到了生物信息中，比如蛋白质工程和基因工程。\n",
    "\n",
    "深度神经网络在蛋白质工程中的用法可以大致分成两类，一类是直接预测突变位点和需要替换的氨基酸。这一类通常分为两步，第一步是知道哪个点不合适，第二步是解决应该用什么氨基酸替换。另一类是给定蛋白质的氨基酸序列或者3D结构，预测突变的效果，比如预测蛋白质-蛋白质相互作用，药物-靶点相互作用，稳定性，热稳定性，能量变化，指定功能等等。第二类方法通常需要跟其他方法结合，比如screening，evolution等得到突变的蛋白质，再用特定模型来预测其突变特性。相比于第一类，这些方法还是比较废时间和人力的。\n",
    "\n",
    "从输入信息来分类的话，sequence-based和structured-based. sequence-based 是利用蛋白质的氨基酸序列来做分析。structured-based是通过蛋白质的蛋白质结构（碳，氢，氧，氮等重要的生物大分子）,通常指3级结构，大部分来源于PDB。蛋白质的三级结构包含了更多的信息且其功能通常由3D结构决定。但这也意味着我们需要处理更多的信息和难度。另外一方面从数据数量上来看，跟氨基酸序列相比，我们所知道的蛋白质结构是相对较少的，随着AlphaFold2的成功，他能预测出基本上所有已知的蛋白质结构，极大的推进了structured-based的研究。但是预测的是有错误率的，在这个基础上进行预测，对准确率还是有影响。另外20个常见氨基酸的排列组合可以说有无数种，但我们一直的常见的或者特定目标的氨基酸序列是少量且固定的，意味着通过氨基酸序列来预测他们的突变位点或者突变效果是有效且可行的。也有一些人同时运用这两种信息来做预测。\n",
    "\n",
    "从深度模型上，目前大部分模型都适用CNN, 包括3DCNN, 2DCNN, residual CNN.\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "739f3731",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
